File: src/main/intellibitz/sted/actions/ItemListenerAction.java
package sted.actions;

import sted.ui.STEDWindow;
import sted.util.MenuHandler;

import javax.swing.*;
import java.awt.event.ActionEvent;
import java.awt.event.ItemEvent;
import java.awt.event.ItemListener;

public class ItemListenerAction
        extends STEDWindowAction
        implements ItemListener {
    public ItemListenerAction() {
        super();
    }

    public void actionPerformed(ActionEvent e) {
    }

    /**
     * Invoked when an item has been selected or deselected by the user. The
     * code written for this method performs the operations that need to occur
     * when an item is selected (or deselected).
     */
    public void itemStateChanged(ItemEvent e) {
        boolean state = true;
        if (ItemEvent.DESELECTED == e.getStateChange()) {
            state = false;
        }
        final Object object = e.getSource();
        final Action action;
        if (AbstractButton.class.isInstance(object)) {
            final STEDWindow stedWindow = getSTEDWindow();
            action = ((AbstractButton) object).getAction();
            AbstractButton button = MenuHandler.getInstance()
                    .getMenuItem((String) action.getValue(Action.NAME));
            if (button != null) {
                button.setSelected(state);
            }
            button = MenuHandler.getInstance()
                    .getToolButton((String) action.getValue(Action.NAME));
            if (button != null) {
                button.setSelected(state);
            }
//            getSTEDWindow().convertSampleText();
        }
    }

}

File: src/main/intellibitz/sted/actions/ItemListenerAction.kt
package sted.actions

import sted.ui.MenuHandler.Companion.instance
import java.awt.event.ActionEvent
import java.awt.event.ItemEvent
import java.awt.event.ItemListener
import javax.swing.AbstractButton

open class ItemListenerAction : STEDWindowAction(), ItemListener {
    override fun actionPerformed(e: ActionEvent) {}

    /**
     * Invoked when an item has been selected or deselected by the user. The
     * code written for this method performs the operations that need to occur
     * when an item is selected (or deselected).
     */
    override fun itemStateChanged(e: ItemEvent) {
        if (e.source is AbstractButton) {
            val name = (e.source as AbstractButton).action?.getValue(NAME) as String
            val state = !(ItemEvent.DESELECTED == e.stateChange)
            instance.getMenuItem(name)?.isSelected = state
            instance.getToolButton(name)?.isSelected = state
//            val stedWindow = sTEDWindow!!
            //            getSTEDWindow().convertSampleText();
        }
    }
}